pg_search/src/postgres/storage/buffer.rs

```diff
@@ -291,11 +291,16 @@ impl BufferMut {
 
     /// Return this [`BufferMut`] instance back to our' Free Space Map, making
     /// it available for future reuse as a new buffer.
-    pub fn return_to_fsm(self, bman: &mut BufferManager) {
+    pub fn return_to_fsm_with_when_recyclable(
+        self,
+        bman: &mut BufferManager,
+        when_recyclable: pg_sys::TransactionId,
```

Comment:
Rather than this being an argument in a bunch of different places, does it make sense as being provided by a global function somewhere, which enterprise would override?